lib/floe/workflow/error_matcher.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Floe
+  class Workflow
+    module ErrorMatcher
+      # note, expecting error_equals from payload to be available
```

Comment:
Since this comment is a requirement, I wouldn't call it a note - I think this can more clearly be written as a comment on the module:


```suggestion
module Floe
  class Workflow
    # Contains methods for common error handling
    #
    # Requires that the including class to have the `error_equals` from the payload to be available.
    module ErrorMatcher
      # note, expecting error_equals from payload to be available
```

or something like that.

Reply:
Or perhaps it's just an addition to the comment for the `match_error?` method specifically.

Reply:
Or perhaps it's a note, but using `@note` yard tag in the `match_error?` method's comment specifically.  I can't really find an `@requires` or `@depends` yard tag.

Reply:
I was able to extract this into the parent so this is no longer needed.

Reply:
that was too much of a leap.
I went back to keeping this PR as just extraction and moving that into #212